src/Tests/Microsoft.DotNet.GenAPI.Tests/CSharpFileBuilderTests.cs

```diff
@@ -154,16 +155,367 @@ public partial struct PublicStruct
 
-                    internal readonly record struct ReadonlyRecordStruct : System.IEquatable<ReadonlyRecordStruct>
+                    internal readonly partial struct ReadonlyRecordStruct : System.IEquatable<ReadonlyRecordStruct>
                     {
+                        [System.Runtime.CompilerServices.CompilerGenerated]
+                        public readonly bool Equals(ReadonlyRecordStruct other) { throw null; }
+
+                        [System.Runtime.CompilerServices.CompilerGenerated]
+                        public override readonly bool Equals(object obj) { throw null; }
+
+                        [System.Runtime.CompilerServices.CompilerGenerated]
+                        public override readonly int GetHashCode() { throw null; }
+
+                        [System.Runtime.CompilerServices.CompilerGenerated]
+                        public static bool operator ==(ReadonlyRecordStruct left, ReadonlyRecordStruct right) { throw null; }
+
+                        [System.Runtime.CompilerServices.CompilerGenerated]
+                        public static bool operator !=(ReadonlyRecordStruct left, ReadonlyRecordStruct right) { throw null; }
+
+                        [System.Runtime.CompilerServices.CompilerGenerated]
+                        public override readonly string ToString() { throw null; }
                     }
 
                     internal readonly partial struct ReadonlyStruct
                     {
                     }
 
-                    internal record struct RecordStruct : System.IEquatable<RecordStruct>
+                    internal partial struct RecordStruct : System.IEquatable<RecordStruct>
+                    {
+                        [System.Runtime.CompilerServices.CompilerGenerated]
+                        public readonly bool Equals(RecordStruct other) { throw null; }
+
+                        [System.Runtime.CompilerServices.CompilerGenerated]
+                        public override readonly bool Equals(object obj) { throw null; }
+
+                        [System.Runtime.CompilerServices.CompilerGenerated]
+                        public override readonly int GetHashCode() { throw null; }
+
+                        [System.Runtime.CompilerServices.CompilerGenerated]
+                        public static bool operator ==(RecordStruct left, RecordStruct right) { throw null; }
+
+                        [System.Runtime.CompilerServices.CompilerGenerated]
+                        public static bool operator !=(RecordStruct left, RecordStruct right) { throw null; }
+
+                        [System.Runtime.CompilerServices.CompilerGenerated]
+                        public override readonly string ToString() { throw null; }
+                    }
+                }
+                """);
+        }
+
+        [Fact]
+        public void TestRecordDeclaration()
+        {
+            RunTest(original: """
+                namespace Foo
+                {
+                    
```

Comment:
Probably unimportant, but do we expect GenAPI to care about extra spaces? Both here and before the end of the namespace.

Reply:
It doesn't this is just input that gets compiled to an assembly.  Gen API never sees it.  I'll clean up extra new lines.